baybe/searchspace/continuous.py

```diff
@@ -383,14 +408,17 @@ def sample_uniform(self, batch_size: int = 1) -> pd.DataFrame:
 
         if not self.parameters:
             return pd.DataFrame(index=pd.RangeIndex(0, batch_size))
-
-        if (
-            len(self.constraints_lin_eq) == 0
-            and len(self.constraints_lin_ineq) == 0
-            and len(self.constraints_cardinality) == 0
-        ):
+        # If the space is completely unconstrained, we can sample from bounds.
+        if not self.is_constrained:
             return self._sample_from_bounds(batch_size, self.comp_rep_bounds.values)
 
+        if self.has_interpoint_constraints:
+            return self._sample_from_polytope_with_interpoint_constraints(
```

Comment:
is it feasible to  keep splittitng the sample function up like that? eg what happens if we have interpoint+cardinality constraints

Reply:
what do you mean exactly?
Is my udnerstadning right that sample from polytope just arranges the cosntraint types accordingly and there are no multiple sample from polytope functions remaining?

Reply:
Exactly, the two functions have been merged, so there is only a single `_sample_from_polytope` function left. This will be called when there are no cardinality constraints

Reply:
Then it would call the `_sample_from_polytope_with_cardinality_constraints` function and interpoint constraints would indeed be ignored. But looking at the code, don't normal constraints also get ignored when there are cardinality constraints?

Reply:
maybe @AdrianSosic can clarify (in a not too long text :) why it is that way

can be accepted if this is intended or required temporarily or as workaround, just making sure this is not accidental

Reply:
@AVHopp: I think the reasoning is not perfectly accurate here. The current logic is:
* At the moment, the only existing nonlinear constraint is the `ContinuousCardinalityConstraint`. Its sampling works by sampling active/inactive sets and then sampling from a polytope on the active set.
* Linear constraints are taken into account by polytope sampling, or to put it differently: polytope sampling IS sampling with linear constraints.

--> Bottom line: all constraints are respected. And we should make sure that stays also true for the interpoint ones. 

Helpful?

Reply:
Resolving here since interpoint constraints and cardinality constraints can now not be used together since this is what was agreed upon internally.